app/src/main/java/com/github/multimatum_team/multimatum/DeadlineDetailsActivity.kt

```diff
@@ -56,8 +49,18 @@ class DeadlineDetailsActivity : AppCompatActivity() {
                 detailView.text = getString(R.string.isAlreadyDue)
             }
             else -> {
-                detailView.text =
-                    getString(R.string.DueInXDays, actualDate.until(date, ChronoUnit.DAYS).toString())
+                val remainingTime = actualDate.until(date, ChronoUnit.DAYS)
+                detailView.text = if (remainingTime <= 0) {
+                    getString(
+                        R.string.DueInXHours,
+                        actualDate.until(date, ChronoUnit.HOURS).toString()
+                    )
+                } else {
+                    getString(
+                        R.string.DueInXDays,
+                        actualDate.until(date, ChronoUnit.DAYS).toString()
+                    )
+                }
             }
         }
```

Comment:
The onCreate function is too long, I think some stuff could be done in private functions